do.sh

```diff
@@ -321,9 +348,23 @@ function mine_data() {
                             | grep ovn-scale | head -3)
     python3 ${topdir}/utils/process-stats.py \
         resource-usage-report-worker.html ${resource_usage_logs}
+    deactivate
+
     popd
 }
 
+function get_tester_ip() {
+    local test_file=$1
+
+    # The tester gets the first IP address in the configured node_net.
+    node_net=$(${ovn_fmn_get} ${test_file} cluster node_net --default=192.16.0.0/16)
```

Comment:
I was thinking about this, and there are a few ways that could work.
1. Use an environment variable to define default values and ensure this same environment variable is used both in the orchestrator and the tester container.
2. Define default values in some file, and refer to this file from `do.sh` and `ovn-tester`.
3. Write a python module similar to `ovn-fake-multinode-tools/get-config-value.py` and put it in `ovn-tester` and define defaults in that module.  Then `do.sh` and `ovn-tester` could both call into this module to get configuration values.
4. Use a template value in `ovn_tester.py` that can be filled in by `do.sh` at install-time.

Of these, I think 1 is clearly the worst. 2 is simple but also a bit clumsy. For me, it's a toss-up whether 3 or 4 is better. I think I would lean more towards option 3 myself. What do you think?

Reply:
Not sure if that is what you meant in 4, but one thing I had in mind previously was:
1. Make ovn-tester fail if at least one variable is not defined in the yml file, i.e. no defaults.
2. Make do.sh generate the full yml from a test scenario yml filling all the undefined values with defaults.
3. Run ovn-tester with a generated fully filled yml.

This will ensure that we do not have defaults defined in multiple places.  Will also be easier to track down undefined/misconfigured values if ovn-tester will just fail and not proceed using unexpected config values.

Reply:
Sounds interesting! It also sounds like something we can do as a follow up in my opinion.  Shall we open an issue to track this?

Reply:
@igsilya Sorry for the ambiguity on number 4. What I meant was that `ovn_tester.py` would have something like this in it:
````
DEFAULT_NODE_NET = {{ default_node_net }}
node_net = netaddr.IPNetwork(cluster_args.get('node_net', DEFAULT_NODE_NET))
````
We might also rename it to `ovn_tester.py.in` to make it clear that it can't be run as-is. Then, when running `do.sh install`, the `do.sh` code will populate `{{ default_node_net }}` with the proper default. This way, the default only is defined in `do.sh`.
It's not my favorite suggestion, but it's something I thought of.

Reply:
@putnopvut Hmm, I'm not sure I like the idea of an `ovn_tester.py.in`.  I think I'd prefer @igsilya's suggestion of do.sh filling in all required input values in the yml we finally pass to ovn-tester.

Reply:
Yes, that's why from the 4 suggestions I came up with, 3 was the one I preferred most. Mine and @igsilya's are more-or-less the same, except that in his, `do.sh` controls the defaults, and in mine, `ovn-tester` controls the defaults.

Reply:
I see, your option "3" is also fine, thanks!

Reply:
I've actually implemented @igsilya's idea at https://github.com/putnopvut/ovn-heater/tree/default_dumper . I was hoping to be able to open a PR here, but github does not allow using a PR as the basis for another PR. Therefore, when I created the PR, it was way larger and more confusing than necessary. I'll hold off and post that PR once this one is merged.